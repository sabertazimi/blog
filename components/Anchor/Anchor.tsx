import type { AnchorProps } from 'antd';
import { Anchor as AntAnchor } from 'antd';

interface Props extends AnchorProps {}

const Anchor = (props: Props): JSX.Element => <AntAnchor {...props} />;

export default Anchor;
