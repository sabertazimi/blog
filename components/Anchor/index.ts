export { Anchor, Link } from './Anchor';
