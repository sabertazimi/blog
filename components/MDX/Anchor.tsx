import { Typography } from 'antd';

interface Props {}

const Anchor = (props: Props): JSX.Element => <Typography.Link {...props} />;

export default Anchor;
