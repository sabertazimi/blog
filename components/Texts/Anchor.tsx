import { cx } from '@components/utils'
import { Typography } from 'antd'
import type { ReactNode } from 'react'
import styles from './Anchor.module.css'

interface Props {
  children?: ReactNode
  className?: string
}

function Anchor({ className, ...props }: Props): JSX.Element {
  return (
    <Typography.Link
      {...props}
      className={cx('mdx-hash-link', className, styles.anchor)}
    />
  )
}

export default Anchor
