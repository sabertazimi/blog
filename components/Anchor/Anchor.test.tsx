import { render } from '@testing-library/react';
import Anchor from './Anchor';

describe('Anchor', () => {
  test('should render correctly (snapshot)', () => {
    const { container } = render(<Anchor />);

    expect(container).toMatchSnapshot();
  });
});
