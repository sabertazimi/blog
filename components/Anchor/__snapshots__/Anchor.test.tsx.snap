// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor should render correctly (snapshot) 1`] = `
<div>
  <a
    class="ant-typography"
  />
</div>
`;
